test/tests/mortar/aux-gap/gap.i

```diff
@@ -0,0 +1,123 @@
+[Mesh]
+  [file]
+    type = FileMeshGenerator
+    file = nodal_normals_test_offset_nonmatching_gap.e
+  []
+  [./primary]
+    input = file
+    type = LowerDBlockFromSidesetGenerator
+    sidesets = '2'
+    new_block_id = '20'
+  [../]
+  [./secondary]
+    input = primary
+    type = LowerDBlockFromSidesetGenerator
+    sidesets = '1'
+    new_block_id = '10'
+  [../]
+[]
+
+[Variables]
+  [./T]
+    block = '1 2'
+  [../]
+  [./lambda]
+    block = '10'
+    use_dual = true
+  [../]
+[]
+
+[AuxVariables]
+  [gap]
+    block = '10'
+  []
+[]
+
+[AuxKernels]
+  [gap]
+    type = WeightedGapAux
+    variable = gap
+    primary_boundary = 2
+    secondary_boundary = 1
+    primary_subdomain = 20
+    secondary_subdomain = 10
+    block = '10'
+  []
+[]
```

Comment:
What happens code-wise if the aux variable interpolation order doesn't coincide with that of the primal variables?

Reply:
See the new test I added. The answer is that it doesn't matter. Since we are doing evaluations on mortar segments where finite element spaces are well defined for every quantity used in the aux computation, we should always be good to go. So match/mismatch with other variable finite element families/orders doesn't matter. You can always use a nodal aux variable that is of order one up to the order of the mesh (so really either order 1 or 2)